From 2948e4db9f223b562b2c064b463d7095d13bc7f8 Mon Sep 17 00:00:00 2001
From: gravis <philippe.lafoucriere@gmail.com>
Date: Wed, 21 Apr 2010 15:29:49 +0200
Subject: [PATCH] make cache_key i18n aware

In I18n website (and all rails app should be considered as being a I18n app), we can't use the cache_key method directly.
The locale must be added manually.
The patch changes the following code :

    - cache "#{user.cache_key}-#{I18n.locale}" do
      ...

into

    - cache user do
      ...

Way cleaner and less error prone.

Ps : no tests found for cache_key ??
---
 activerecord/lib/active_record/base.rb |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/activerecord/lib/active_record/base.rb b/activerecord/lib/active_record/base.rb
index fd24dcd..d0f5209 100755
--- a/activerecord/lib/active_record/base.rb
+++ b/activerecord/lib/active_record/base.rb
@@ -1601,17 +1601,17 @@ module ActiveRecord #:nodoc:
       #
       # ==== Examples
       #
-      #   Product.new.cache_key     # => "products/new"
-      #   Product.find(5).cache_key # => "products/5" (updated_at not available)
-      #   Person.find(5).cache_key  # => "people/5-20071224150000" (updated_at available)
+      #   Product.new.cache_key     # => "products/new-en"
+      #   Product.find(5).cache_key # => "products/5-en" (updated_at not available)
+      #   Person.find(5).cache_key  # => "people/5-20071224150000-en" (updated_at available)
       def cache_key
         case
         when new_record?
-          "#{self.class.model_name.cache_key}/new"
+          "#{self.class.model_name.cache_key}/new-#{I18n.locale}"
         when timestamp = self[:updated_at]
-          "#{self.class.model_name.cache_key}/#{id}-#{timestamp.to_s(:number)}"
+          "#{self.class.model_name.cache_key}/#{id}-#{timestamp.to_s(:number)}-#{I18n.locale}"
         else
-          "#{self.class.model_name.cache_key}/#{id}"
+          "#{self.class.model_name.cache_key}/#{id}-#{I18n.locale}"
         end
       end
 
-- 
1.7.0.4

